src/emulator/hubExport.ts

```diff
@@ -0,0 +1,68 @@
+import * as path from "path";
+import * as fs from "fs";
+
+import * as utils from "../utils";
+import * as api from "../api";
+import { IMPORT_EXPORT_EMULATORS, Emulators, ALL_EMULATORS } from "./types";
+import { EmulatorRegistry } from "./registry";
+import { FirebaseError } from "../error";
+
+export interface ExportMetadata {
+  firestore?: string;
+}
+
+export class HubExport {
+  static METADATA_FILE_NAME = "metadata.json";
+
+  constructor(private projectId: string, private exportPath: string) {}
+
+  public async exportAll(): Promise<void> {
+    const toExport = ALL_EMULATORS.filter(this.shouldExport);
+    if (toExport.length === 0) {
+      throw new FirebaseError("No running emulators support import/export.");
+    }
+
+    // TODO(samstern): Once we add other emulators, we have to deal with the fact that
+    // there may be an existing metadata file and it may only partially overlap with
+    // the new one.
```

Comment:
In that case shall we put CLI versions into `ExportMetadata` as well? I mean, we don't have to check it right now, but it's great to have that option available in the future.

Reply:
Good idea.